molecule/shared/substrate/lxd/create.yml

```diff
@@ -0,0 +1,84 @@
+---
+- name: Manage LXD containers
+  hosts: localhost
+  gather_facts: false
+  connection: local
+  vars:
+    molecule_inventory:
+      molecule:
+        hosts: {}
+    molecule_yml:
+      platforms:
+        - name: instance
+          image: ubuntu:20.04
+          user: ansible
+
+  tasks:
+    - name: Ensure LXD is installed
+      ansible.builtin.command:
+        cmd: snap install lxd
+      become: true
+
+    - name: Initialize LXD (if not already initialized)
+      command: lxd init --auto
+      args:
+        creates: /var/lib/lxd/lxd.db
+      become: true
+
+    - name: Check if LXD container exists
+      command: lxc list "{{ platform['name'] }}" --format=json
+      register: lxd_container_info
+      ignore_errors: true
+      loop: "{{ molecule_yml['platforms'] }}"
+      loop_control:
+        loop_var: platform
+
+    - name: Debug container existence
+      ansible.builtin.debug:
+        var: lxd_container_info
+
+    - name: Create LXD containers if not exists
+      community.general.lxd_container:
+        name: "{{ platform['name'] }}"
+        state: started
+        source:
+          type: image
+          mode: pull
+          server: https://cloud-images.ubuntu.com/releases
+          protocol: simplestreams
+          alias: "f"
```

Comment:
I would recommend adding the following, to leverage the `cloud-init` that's used to stand-up vms in this collection (probably why you need to add so many preconditions)

```yml
        config:
          cloud-init.user-data: "{{ lookup('ansible.builtin.template', '../../../../depin/core/playbooks/templates/user-data.yml.j2 ') }}"
        wait_for_ipv4_addresses: true
        wait_for_container: true
        timeout: 300
```

Also the last couple lines here ensures the VM is online before preceeding

Reply:
used

```
  vars:
    molecule_inventory:
      molecule:
        hosts: {}
    molecule_yml:
      platforms:
        - name: instance
          image_fingerprint: "noble"
          architecture: x86_64
          user: ansible
```